onboard/catkin_ws/src/controls/scripts/desired_state.py

```diff
@@ -99,34 +114,16 @@ def run(self):
                 utils.publish_data_dictionary(self.pub_pos, utils.get_axes(), self.pose)
                 self.pose = None
 
-            elif self.powers:
-                if self.last_powers is None or self.powers != self.last_powers:
-                    self.enable_loops()
-                    # Hold Position on the current state
-                    self.hold = dict(self.state)
-                    self.last_powers = dict(self.powers)
-
-                # Nonzero entries bypass PID
-                # If any nonzero xy power, disable those position pid loops
-                if self.powers['x'] != 0 or self.powers['y'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['x', 'y'], False)
-
-                # If any nonzero roll or pitch power, disable those position pid loops
-                if self.powers['roll'] != 0 or self.powers['pitch'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['roll', 'pitch'], False)
-
-                # If nonzero yaw power, disable those position pid loops
-                if self.powers['yaw'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['yaw'], False)
-
-                # If any nonzero z power, disable those position pid loops
-                if self.powers['z'] != 0:
-                    utils.publish_data_constant(self.pub_pos_enable, ['z'], False)
-
-                utils.publish_data_dictionary(self.pub_power, utils.get_axes(), self.powers)
-                # Publish current state to the desired state for PID
-                utils.publish_data_dictionary(self.pub_pos, utils.get_axes(), self.hold)
-                self.powers = None
+            elif self.twist:
+                self.enable_loops()
+                self.disable_pos_loop()
+                utils.publish_data_dictionary(self.pub_vel, utils.get_axes(), self.twist)
+                self.twist = None
+
+            elif self.power:
+                self.disable_loops()
+                utils.publish_data_dictionary(self.pub_power, utils.get_axes(), self.power)
+                self.power = None
```

Comment:
This code has the side effect of no longer holding state on power control. Is this intended? I think joystick would prefer if you did maintain state like before, cause who wants try to figure out the exact amount by which to hold the joystick down for the robot to not float straight up.